const Solicitud = require('../models/solicitud')

const list = async (req, res) => {
  const response = await Solicitud.findAll({
    include: ['Aspirante', 'Trabajador', 'Animal', 'Visita']
  })
    .then(function (data) {
      const res = { success: true, data: data }
      return res
    })
    .catch(error => {
      const res = { success: false, error: error }
      return res
    })
  res.json(response)
}

const create = async (req, res) => {
  try {
    const response = await Solicitud.create({
      tiempo: req.body.tiempo,
      AspiranteId: req.body.AspiranteId,
      AnimalId: req.body.AnimalId,
      VisitaId: req.body.VisitaId,
      TrabajadorId: req.body.TrabajadorId
    })
      .then(function (data) {
        const res = {
          success: true,
          data: data,
          message: 'creado exitosamente'
        }
        return res
      })
      .catch(error => {
        const res = { success: false, error: error }
        return res
      })
    res.json(response)
  } catch (e) {
    console.log(e)
  }
}

const update = async (req, res) => {
  try {
    const { id } = req.params

    const response = await Solicitud.update(
      {
        tiempo: req.body.tiempo,
        AspiranteId: req.body.AspiranteId,
        AnimalId: req.body.AnimalId,
        VisitaId: req.body.VisitaId,
        TrabajadorId: req.body.TrabajadorId
      },
      {
        where: { id: id }
      }
    )
      .then(function (data) {
        const res = {
          success: true,
          data: data,
          message: 'actualizado exitosamente'
        }
        return res
      })
      .catch(error => {
        const res = { success: false, error: error }
        return res
      })
    res.json(response)
  } catch (e) {
    console.log(e)
  }
}

const getOne = async (req, res) => {
  try {
    const { id } = req.params

    const response = await Solicitud.findAll({
      where: { id: id }
    })
      .then(function (data) {
        const res = { success: true, data: data }
        return res
      })
      .catch(error => {
        const res = { success: false, error: error }
        return res
      })
    res.json(response)
  } catch (e) {
    console.log(e)
  }
}

const deleteOne = async (req, res) => {
  try {
    const { id } = req.params

    const response = await Solicitud.destroy({
      where: { id: id }
    })
      .then(function (data) {
        const res = {
          success: true,
          data: data,
          message: 'borrado exitosamente'
        }
        return res
      })
      .catch(error => {
        const res = { success: false, error: error }
        return res
      })
    res.json(response)
  } catch (e) {
    console.log(e)
  }
}

module.exports = {
  list,
  create,
  update,
  getOne,
  deleteOne
}
